### DISCUSSION TITLE
[v5] user.preload is not a function

### DISCUSSION THREAD

**Discussion by kempsteven** [adonisjs/core#2183] [Help]
sooo i have a model like this:

```js
export default class LeaveApplication extends BaseModel {
  @column({ isPrimary: true })
  public id: string

  @column()
  public user_id: string

  @belongsTo(() => User, {
    foreignKey: 'user_id',
  })
  public user: BelongsTo<typeof User>

  @column.date({
    serialize: (value?: DateTime) => {
      return value ? value.toISODate() : value
    },
  })
  public date_from: DateTime

  @column.date({
    serialize: (value?: DateTime) => {
      return value ? value.toISODate() : value
    },
  })
  public date_to: DateTime

  @column.dateTime({ autoCreate: true })
  public created_at: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updated_at: DateTime
}
```

I had to serialize the model so I could customize the value of `date_to` and `date_from`,
so this is the query I have made:

```js
    const leaveApplications = await LeaveApplication
      .query()
      .preload('user')
      .orderBy('date_from', 'asc')

    return leaveApplications.map((x) => x.serialize())
```

it did what it had to do, but i get an error `TypeError: user.preload is not a function`,
any reason why this happens?

If I did not serialize it works good.
Also is this the best way to serialize an array of results, not sure if using .map is the best way to do it.

**Comment by thetutlage** [comment]
Can you share the complete error stack? 

> Also is this the best way to serialize an array of results, not sure if using .map is the best way to do it.

Mind explaining why you think `.map` is not the best way?
